scene/resources/packed_scene.cpp

```diff
@@ -1849,6 +1849,44 @@ void SceneState::add_editable_instance(const NodePath &p_path) {
 	editable_instances.push_back(p_path);
 }
 
+bool SceneState::remove_group_references(const StringName &p_name) {
+	bool edited = false;
+	for (int node_idx = 0; node_idx < get_node_count(); node_idx++) {
+		for (int group_idx = 0; group_idx < nodes[node_idx].groups.size(); group_idx++) {
+			if (names[nodes[node_idx].groups[group_idx]] == p_name) {
+				nodes.write[node_idx].groups.remove_at(group_idx);
+				edited = true;
+				break;
+			}
+		}
+	}
+	return edited;
+}
+
+bool SceneState::rename_group_references(const StringName &p_old_name, const StringName &p_new_name) {
+	bool edited = false;
+	for (int node_idx = 0; node_idx < get_node_count(); node_idx++) {
+		for (int group_idx = 0; group_idx < nodes[node_idx].groups.size(); group_idx++) {
+			if (names[nodes[node_idx].groups[group_idx]] == p_old_name) {
+				names.set(nodes[node_idx].groups[group_idx], p_new_name);
+				edited = true;
+				break;
+			}
+		}
+	}
+	return edited;
+}
+
+HashSet<StringName> SceneState::get_all_groups() {
+	HashSet<StringName> ret;
+	for (int node_idx = 0; node_idx < get_node_count(); node_idx++) {
+		for (int group_idx = 0; group_idx < nodes[node_idx].groups.size(); group_idx++) {
+			ret.insert(names[nodes[node_idx].groups[group_idx]]);
+		}
+	}
```

Comment:
```suggestion
	for (const NodeData &node : nodes) {
		for (const int &group : node.groups) {
			ret.insert(names[group]);
		}
	}
```
Prefer range iterators where possible (same in other methods).